amqprs/src/net/split_connection.rs

```diff
@@ -131,8 +131,8 @@ impl SplitConnection {
     }
 
     #[cfg(feature = "tls")]
-    pub async fn open_tls(addr: &str, domain: &str, connector: &TlsConnector) -> Result<Self> {
-        let domain = rustls::ServerName::try_from(domain)
+    pub async fn open_tls(addr: &str, domain: String, connector: &TlsConnector) -> Result<Self> {
```

Comment:
no need to change to `String`, `&str` is most flexible type

Reply:
Reverted by https://github.com/gftea/amqprs/pull/141/commits/0b5310bbc9cf2557f5c337a350de46a0fa8481ba